src/components/LMComment/index.tsx

```diff
@@ -0,0 +1,253 @@
+import {View, TouchableOpacity, StyleSheet, FlatList} from 'react-native';
+import React, {useCallback, useEffect, useState} from 'react';
+import LMText from '../../base/LMText';
+import layout from '../../utils/layout';
+import {LMCommentProps} from './types';
+import LMButton from '../../base/LMButton';
+import LMInputText from '../../base/LMInputText';
+import {timeStamp} from '../../utils';
+import LMIcon from '../../base/LMIcon';
+import {PARENT_LEVEL_COMMENT} from '../../constants/strings';
+import {FlashList} from '@shopify/flash-list';
+
+const LMComment = ({
+  likeIconButton,
+  likeTextButton,
+  comment,
+  onTapViewMore,
+  commentMaxLines,
+  menuIcon,
+  commentUserNameStyle,
+  commentContentProps,
+  showMoreProps,
+  replyTextProps,
+  repliesCountTextStyle,
+  timeStampStyle,
+  replyInputFieldProps,
+  viewMoreRepliesProps,
+}: LMCommentProps) => {
+  const MAX_LINES = commentMaxLines ? commentMaxLines : 3;
+  const [showText, setShowText] = useState(false);
+  const [numberOfLines, setNumberOfLines] = useState<number>();
+  const [showMoreButton, setShowMoreButton] = useState(false);
+  const [showReplies, setShowReplies] = useState(false);
+  const [showReplyInput, setShowReplyInput] = useState(false);
+  const [repliesArray, setRepliesArray] = useState(comment.replies);
+
+  // this handles the show more functionality
+  const onTextLayout = useCallback(
+    (e: any) => {
+      if (e.nativeEvent.lines.length > MAX_LINES && !showText) {
+        setShowMoreButton(true);
+        setNumberOfLines(MAX_LINES);
+      }
+    },
+    [showText],
+  );
+
+  // this handles the visiblity of whole comment content and trimmed text upto maximum line
+  useEffect(() => {
+    if (showMoreButton) {
+      setNumberOfLines(showText ? undefined : MAX_LINES);
+    }
+  }, [showText, showMoreButton]);
+
+  //creating content props as per customization
+  const updatedContentProps = commentContentProps
+    ? commentContentProps
+    : {
+        text: comment.text,
+        onTextLayout: (e: any) => onTextLayout(e),
```

Comment:
why `e`? it should be named as `text` or `inputText`

Reply:
It is basically an event that occurs whenever the layout of the text changes. This occur on load and return some native properties of the text like in how many lines the text will be rendered.